src/main/drivers/bus_spi_impl.h

```diff
@@ -46,7 +46,11 @@ typedef struct spiHardware_s {
     SPI_TypeDef *reg;
     spiPinDef_t sckPins[MAX_SPI_PIN_SEL];
     spiPinDef_t misoPins[MAX_SPI_PIN_SEL];
+#if defined(AT32F4)
+    spiPinDef_t mosiPins[MAX_SPI_PIN_SEL+1];
```

Comment:
```suggestion
    spiPinDef_t mosiPins[MAX_SPI_PIN_SEL];
```

IMO we should change line 27 - 29 to
```
#elif defined(STM32F7)
#define MAX_SPI_PIN_SEL 4
#elif defined(STM32H7 || defined(AT32F4)
#define MAX_SPI_PIN_SEL 5
```

Reply:
Yes this would no longer require the addition of code block using index + 1.
Created a new PR in #14324

Reply:
> 是的，这不再需要使用 index + 1 添加代码块。在 #14324 中创建新的 PR

I tried this on 4.5.1, but it didn't work, why?

Reply:
It is based on 4.6 code, if verified we can back port to 4.5.3